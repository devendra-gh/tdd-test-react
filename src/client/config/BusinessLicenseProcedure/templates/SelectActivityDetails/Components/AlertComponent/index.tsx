import React, { useEffect } from 'react';
import Alert from '@tamm/ui-lib-v2-alert';
import { IVariables } from '@tamm/app-composer';

function AlertComponent(props: IVariables) {
  const { i18n, formSelectActivity } = props;
  const { showError, alertText, alertStatus, inputType } = formSelectActivity;

  useEffect(() => {
    if (alertStatus === 'error') {
      props.history.push('/business-licence-procedure/error-page');
    }
  }, [alertStatus]);

  const getAlert = () => {
    if (inputType === 'CATEGORY' && alertText === 'errorMessage.noData')
      return i18n('errorMessage.noData.category');
    return i18n(alertText);
  };
  return (
    <div className="alert-container">
      {showError && <Alert message={getAlert()} status={alertStatus} />}
    </div>
  );
}

export default AlertComponent;
