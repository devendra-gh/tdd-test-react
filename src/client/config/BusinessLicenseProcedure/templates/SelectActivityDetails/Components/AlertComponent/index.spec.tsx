import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router';
import { IVariables } from '@tamm/app-composer';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import AlertComponent from './index';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('AlertComponent', () => {
  let props: IVariables;

  beforeEach(() => {
    props = {
      history: {
        push: jest.fn(),
      },
      actions: {
        stepsStatus: {
          update: jest.fn(),
        },
        formSelectActivity: {
          update: jest.fn(),
        },
      },
      getInitialState: jest.fn(),
      formSelectActivity: {
        activities: [
          {
            activityCode: 'activityCode',
            activity: 'businessActivity.activity',
            activityAr: 'businessActivity.activityAr',
          },
        ],
        showTable: true,
        category: 'category',
        subCategory: 'subCategory',
        inputType: 'CATEGORY',
        loading: true,
        showError: true,
        alertText: 'alertText',
        alertStatus: 'info',
      },
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      i18n: (key: string) => key,
      title: 'FormTemplate',
      subTitle: 'subTitle',
      description: 'desc',
      locale: 'en',
      currentStep: 'economic_licence',
      currentSubStep: 'payment',
      buttons: [
        {
          label: 'btn',
          onClick: jest.fn(),
        },
      ],
      steps: [],
      stepStatus: {
        economic_name: 'economic_name',
        'economic_name.payment': 'economic_name.payment',
      },
      list: {
        labelHeading: 'label',
        valueHeading: 'value',
        listDetails: {
          label: 'label',
          value: 'value',
        },
      },
      totalSection: 1000,
    };
  });

  it('should render listOfActivities when showTable true - english', async () => {
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <AlertComponent {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(getByText('alertText')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render listOfActivities when showTable true - english with error', async () => {
    props.formSelectActivity.alertStatus = 'error';
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <AlertComponent {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(getByText('alertText')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render listOfActivities when showTable true - english', async () => {
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <AlertComponent {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(getByText('alertText')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render listOfActivities when showTable true - english', async () => {
    props.formSelectActivity.alertText = 'errorMessage.noData';
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <AlertComponent {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );
    expect(getByText('errorMessage.noData.category')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
