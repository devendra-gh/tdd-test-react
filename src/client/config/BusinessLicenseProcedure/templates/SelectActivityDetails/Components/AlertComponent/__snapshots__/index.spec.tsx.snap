// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertComponent should render listOfActivities when showTable true - english 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="alert-container"
  >
    <div
      class="ui-lib-alert ui-lib-alert_info"
    >
      <svg
        class="ui-lib-alert__icon ui-lib-icon"
        data-key=""
        height="24"
        tabindex="0"
        version="1.1"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <path
          d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
        />
      </svg>
      <div
        class="ui-lib-alert__message"
      >
        alertText
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertComponent should render listOfActivities when showTable true - english 2`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="alert-container"
  >
    <div
      class="ui-lib-alert ui-lib-alert_info"
    >
      <svg
        class="ui-lib-alert__icon ui-lib-icon"
        data-key=""
        height="24"
        tabindex="0"
        version="1.1"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <path
          d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
        />
      </svg>
      <div
        class="ui-lib-alert__message"
      >
        alertText
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertComponent should render listOfActivities when showTable true - english 3`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="alert-container"
  >
    <div
      class="ui-lib-alert ui-lib-alert_info"
    >
      <svg
        class="ui-lib-alert__icon ui-lib-icon"
        data-key=""
        height="24"
        tabindex="0"
        version="1.1"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <path
          d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
        />
      </svg>
      <div
        class="ui-lib-alert__message"
      >
        errorMessage.noData.category
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertComponent should render listOfActivities when showTable true - english with error 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="alert-container"
  >
    <div
      class="ui-lib-alert ui-lib-alert_error"
    >
      <svg
        class="ui-lib-alert__icon ui-lib-icon"
        data-key=""
        height="24"
        tabindex="0"
        version="1.1"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title />
        <path
          d="M20.488,3.515a11.994,11.994,0,1,0,0,16.97A12.014,12.014,0,0,0,20.488,3.515ZM16.709,15.279A1,1,0,0,1,15.3,16.694l-3.11-3.11a.252.252,0,0,0-.354,0l-3.109,3.11a1.019,1.019,0,0,1-1.415,0,1,1,0,0,1,0-1.415l3.11-3.109a.249.249,0,0,0,0-.353l-3.11-3.11A1,1,0,0,1,8.722,7.292l3.109,3.11a.25.25,0,0,0,.354,0l3.11-3.11a1,1,0,0,1,1.414,1.415l-3.11,3.11a.25.25,0,0,0,0,.353Z"
        />
      </svg>
      <div
        class="ui-lib-alert__message"
      >
        alertText
      </div>
    </div>
  </div>
</div>
`;
